src/Core/Context/CurrentContext.cs

```diff
@@ -361,9 +347,20 @@ public async Task<bool> DeleteAssignedCollections(Guid orgId)
 
     public async Task<bool> ViewAssignedCollections(Guid orgId)
     {
-        return await CreateNewCollections(orgId) // Required to display the existing collections under which the new collection can be nested
-               || await EditAssignedCollections(orgId)
-               || await DeleteAssignedCollections(orgId);
+        /*
+         * Required to display the existing collections under which the new collection can be nested.
+         * Owner, Admin, Manager, and Provider checks are handled via the EditAssigned/DeleteAssigned context calls.
+         * This entire method will be moved to the ControllerAuthorizationHandler in the future
+         */
+        var canCreateNewCollections = false;
+        var org = GetOrganization(orgId);
+        if (org != null)
+        {
+            canCreateNewCollections = !org.LimitCollectionCdOwnerAdmin || org.Permissions.CreateNewCollections;
```

Comment:
This doesn't properly replace the `CurrentContext.CreateNewCollections` method (or replicate the new check in the auth handler). The second half of the boolean is only checking the custom permission `CreateNewCollections`, not whether the user is an owner/admin/provider.

Reply:
This is correct - however, because it is paired with two other current context permission checks using the `OR` operator, the remaining owner/admin/manager/provider checks are handled there **first** and if nothing passes above, the remaining collection management and `CreateNewCollections` permission is checked.

Reply:
I accept that!